### PR TITLE
Module, block and enumerable

### PR DESCRIPTION
**Pull request by mntedev**
**We completed the following tasks. :+1:** 

:heavy_check_mark:  Used Ruby module
:heavy_check_mark:  Used block and block best practice
:heavy_check_mark:  Used Ruby module in Ruby class
:heavy_check_mark:  Used Ruby best practice.
:heavy_check_mark: Implemented enumerable namely all, any, and filter.
:heavy_check_mark:  Checked for linters error.

### CODE REVIEW

File: module.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    end
+    p false
+  end
+
+  def filter
+    ary = []
+    list.filter do |item|
+      ary.push(item) if yield item
+    end
+    p ary
```
Comment (DammyShittu) [inline module.rb]: - [ ] These methods should **return** the results, not print them.

File: my_list.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+require_relative 'module'
+
+class MyList
+  include MyEnumerable
+  def initialize(*list)
+    @list = list
+  end
+end
```
Comment (DammyShittu) [inline my_list.rb]: - [ ] The MyList class should contain your own functionality of the Ruby `#each` method. And the enumerables in the module.rb file should use this implementation. Kindly add it here.

  ```
  def my_each
    return to_enum(:each) unless block_given?

    counter = 0
    while counter < @list.length
      yield(@list[counter])
      counter += 1
    end
    @list
  end
  ```

File: module.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  end
```
Comment (DammyShittu) [inline module.rb]: - [ ] These enumerables should be implemented using the "my_each" method created in the MyList class. For example

  ```
  module MyEnumerable
    def all?
      my_each { |item| return false unless yield item }
      
      true
    end
  
    def any?
      my_each { |item| return true if yield item }
      false
    end
  
    def filter
      filter_array = []
      my_each { |item| filter_array << item if yield item }
  
      filter_array
    end
  end
  ```

File: module.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
+module MyEnumerable
+  def list
+    @list ||= []
+  end
```
Comment (DammyShittu) [inline module.rb]: - [ ] This method would be unnecessary. We should be using the `@list` variable passed from the MyList class anyways. Please remove it

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @worashf @masangana,

## Highlights
✔️ Descriptive PR
✔️ Passing linters

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: module.rb [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  def all?
+    my_each { |item| return true if yield item }
+    true
+  end
+
+  def any?
+    my_each { |item| return false unless yield item }
+    false
+  end
+
```
Comment (DammyShittu) [inline module.rb]: - [ ] You implemented the changes wrongly. The tests below should return, `false`, `false`, `false`, `true` respectively but that's not the case from the screenshot below

  ```
  list = MyList.new(1, 2, 3, 4, 5, 6, 7, 8, 9, 10)
  p list
  
  p list.all? { |e| e < 5 }
  p list.all? { |e| e > 5 }
  
  p list.any?(&:zero?)
  p list.any? { |e| e == 5 }
  ```

  list

  Kindly pay close attention to the combination of if/unless and the return statements in the enumerable snippet I suggested. The `all?` method should be this:
  ```
  def all?
    my_each { |item| return false unless yield item }
    true
  end
  ```
  The `any?` method should be this:
  ```
  def any?
    my_each { |item| return true if yield item }
    false
  end
  ```

File: README.md [hunk 6]
```diff
@@ -2,55 +2,57 @@
 
 # Project Name
```
Comment (DammyShittu) [inline README.md:3]: - **[OPTIONAL]** This was missed in the previous review, however I strong suggest your give your readme a proper title

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @worashf @masangana,

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 6 above)
Comment (masangana) [reply README.md:3]: solved

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @Team

Your project is complete! Great job for making the changes 🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request
- Validations correctly implemented
- No errors

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**